core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -115,6 +121,10 @@ private[scheduler] class BlacklistTracker (
         logInfo(s"Removing nodes $nodesToUnblacklist from blacklist because the blacklist " +
           s"has timed out")
         nodeIdToBlacklistExpiryTime --= nodesToUnblacklist
+        nodesToUnblacklist.foreach { node =>
+          nodeIdToBlacklistExpiryTime.remove(node)
```

Comment:
Yes, that's an error. The foreach block is correct so that we can send an event on the following line.